### PR TITLE
[PowerPC] Add support for -mcpu=pwr11 / -mtune=pwr11

### PR DESCRIPTION
**Pull request by azhan92**
This PR adds support for -mcpu=pwr11/power11 and -mtune=pwr11/power11 in clang and llvm.

### CODE REVIEW

File: llvm/lib/Target/PowerPC/PPC.td [hunk 1]
```diff
@@ -672,6 +685,8 @@ def : ProcessorModel<"pwr7", P7Model, ProcessorFeatures.P7Features>;
 def : ProcessorModel<"pwr8", P8Model, ProcessorFeatures.P8Features>;
 def : ProcessorModel<"pwr9", P9Model, ProcessorFeatures.P9Features>;
 def : ProcessorModel<"pwr10", P10Model, ProcessorFeatures.P10Features>;
+// No scheduler model yet.
```
Comment (chenzheng1030) [inline llvm/lib/Target/PowerPC/PPC.td:688]: Is it ok we just use P10Model for P11?

**Review by chenzheng1030 — COMMENTED** [review]
Basically you need to handle P11 for all the places where P10 is handled, seems you are missing below places:
1: we need handling in `LoadOpcodesForSpill` and `StoreOpcodesForSpill` for P11. Even for now P11 may already use the model of P10, we need at least some comments there to say P11 is reusing the same model with P10.
2: in `PPCISelLowering.cpp`, we also need to handle P11 for function `combineMUL` and `getPrefLoopAlignment`.
3: in `Host.cpp`, the funciton `getHostCPUName()`

**Review by chenzheng1030 — COMMENTED** [review]
We also need more tests like:
1: `llvm/test/CodeGen/PowerPC/check-cpu.ll`
2: at least one test in the backend like `llvm/test/CodeGen/PowerPC/p10-constants.ll`?

File: llvm/test/CodeGen/PowerPC/mma-acc-spill-p11.ll [hunk 2]
```diff
@@ -0,0 +1,122 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; ; This test is a copy of mma-acc-spill.ll except that it uses mcpu=pwr11.
+; RUN: llc -verify-machineinstrs -mtriple=powerpc64le-unknown-linux-gnu \
+; RUN:   -mcpu=pwr11 -ppc-asm-full-reg-names -disable-auto-paired-vec-st=false \
```
Comment (daltenty) [inline llvm/test/CodeGen/PowerPC/mma-acc-spill-p11.ll:4]: I don't quite follow why this test got versioned. If I understand correctly, the codegen should be identical (at least for now), so I think we should just add the extra run lines to the existing test if anything.

**Review by daltenty — APPROVED** [review]
LGTM, thanks!

File: llvm/lib/Target/PowerPC/PPCInstrInfo.cpp [hunk 3]
```diff
@@ -3485,9 +3485,11 @@ unsigned PPCInstrInfo::getSpillTarget() const {
   // With P10, we may need to spill paired vector registers or accumulator
   // registers. MMA implies paired vectors, so we can just check that.
   bool IsP10Variant = Subtarget.isISA3_1() || Subtarget.pairedVectorMemops();
-  return Subtarget.isISAFuture() ? 3 : IsP10Variant ?
-                                   2 : Subtarget.hasP9Vector() ?
-                                   1 : 0;
+  // P11 uses the P10 target.
+  return Subtarget.isISAFuture()   ? 3
```
Comment (chenzheng1030) [inline llvm/lib/Target/PowerPC/PPCInstrInfo.cpp:3489]: nit: is this formatted by clang-format? Maybe we should not change these lines as you are not changing them.

**Review by chenzheng1030 — APPROVED** [review]
LGTM except one nit. Thanks very much for adding this support.

File: llvm/lib/Target/PowerPC/PPCInstrInfo.cpp (hunk 3 above)
Comment (azhan92) [reply llvm/lib/Target/PowerPC/PPCInstrInfo.cpp:3489]: Oh yes, it was formatted by clang-format. I can remove the changes

**Pull request merged**